projects/igniteui-angular/src/lib/grid/grid.component.ts

```diff
@@ -448,6 +457,97 @@ export class IgxGridComponent implements OnInit, OnDestroy, AfterContentInit, Af
         this.cdr.detectChanges();
     }
 
+    /* Toolbar related definitions */
+
+    private _showToolbar = false;
+    private _exportExcel = false;
+    private _exportCsv = false;
+    private _toolbarTitle: string = null;
+    private _exportText: string = null;
+    private _exportExcelText: string = null;
+    private _exportCsvText: string = null;
+
+    @ViewChild('toolbar', { read: IgxGridToolbarComponent })
+    public toolbar: IgxGridToolbarComponent = null;
+
+    @ViewChild('toolbar', { read: ElementRef })
+    private toolbarHtml: ElementRef = null;
+
+    @Input()
+    public get showToolbar(): boolean {
+        return this._showToolbar;
+    }
+
+    public set showToolbar(newValue: boolean) {
+        this._showToolbar = newValue;
```

Comment:
Make sure that the value has actually changed here by adding a this._showToolbar !== newValue check. This applies to all setters below.

Reply:
Checks added to all toolbar related setters.